ledger-tool/src/main.rs

```diff
@@ -1598,7 +1614,92 @@ fn main() {
                         },
                     );
 
-                    let process_options = parse_process_options(&ledger_path, arg_matches);
+                    let mut process_options = parse_process_options(&ledger_path, arg_matches);
+
+                    let include_bank = match arg_matches.value_of("verify_slots_details").unwrap() {
+                        "none" => false,
+                        "bank" => true,
+                        _ => unreachable!(),
+                    };
+
+                    let (slot_callback, record_slots_file, recorded_slots) = if let Some(filename) =
+                        arg_matches.value_of_os("verify_slots")
```

Comment:
Just confirming the usage here around `verify_slots`:
- Arg not passed ==> skip all of this
- Arg passed + file exists ==> file contains details to check against, enter "check mode"
- Arg passed + file does not exist ==> enter "record mode"

And then `verify_slot_details` controls the "verbosity", ie, what level of detail we put in ... either the full bank + accounts OR just the hashes

Reply:
Yes, that's right. I've added this in a comment (above the cli processing)

Reply:
Gotcha. I think I'd be more be more keen to having separate flags. So, something like:
```
--record-bank-details file
    - We could potentially make file optional and write to default location 
--record-bank-details-config config
    - Not sure config is the write word here, but some word to note that this is modifying the output
--verify-bank-details file
    - file is required and where details will be read from to check against
```
And we can enforce some sanity with CLAP:
- `--record-bank-details-config` requires `--record-bank-details`
- `--verify-bank-details` conflicts with `--record-bank-details` (ie cannot specify both)

Some other random thoughts:
- I think we should be a little more verbose with `config` options, maybe something like `hash-only` and `accounts`
- I go back and forth on whether to use `bank` or `slot` here; what do you think?
- There is an existing flag / some functionality with `--write-bank-file`. That flag dumps only the last slot (ie `--halt-at-slot`), and also dumps the file into a specific directory.

The pre-existing flag and the flag(s) you're looking to add are similar enough that I think we should try to unify them.

Reply:
Separate flags is much nicer, did not occur to me so thanks for that.

There is an issue with clap 2.33 that any argument with a default value does not work with `.conflicts_with()`. It is a shame that we're on ancient version of clap. I have not implemented the conflicts because of this.

Like you, today I also flip flopped on bank & slot. I don't feel particularly strongly on either. However, the slot callback in we use in the block store processor is on a frozen bank/slot: we work on frozen banks, i.e. slots. 

I am inclined to remove the `--write-bank-file` option, but I don't know if there is some use case I am missing. Maybe something for another PR.

It's definitely much cleaner after your comments. Please let me know what you think. Thanks.

Reply:
> There is an issue with clap 2.33 that any argument with a default value does not work with .conflicts_with(). It is a shame that we're on ancient version of clap. I have not implemented the conflicts because of this.

Ugh, dang it but makes sense

> Maybe something for another PR.

Yeah, fair enough. We're still a little ways out from cutting a new branch, so plenty of time to figure out / cleanup without the intermediate getting into a release.

I can double check cases of the existing flag, and then deprecate it. I think for at least one release, we can detect that flag was passed and give a friendly error to use the new flags